vmtkScripts/contrib/vmtknumpytoimage.py

```diff
@@ -0,0 +1,88 @@
+#!/usr/bin/env python
+
+## Program:   VMTK
+## Module:    $RCSfile: vmtknumpytoimage.py,v $
+## Language:  Python
+## Date:      June 10, 2017
+## Version:   1.4
+
+##   Copyright (c) Richard Izzo, Luca Antiga, David Steinman. All rights reserved.
+##   See LICENCE file for details.
+
+##      This software is distributed WITHOUT ANY WARRANTY; without even
+##      the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR
+##      PURPOSE.  See the above copyright notices for more information.
+
+## Note: this class was contributed by
+##       Richard Izzo (Github @rlizzo)
+##       University at Buffalo
+##       The Jacobs Institute
+
+from __future__ import absolute_import  # NEEDS TO STAY AS TOP LEVEL MODULE FOR Py2-3 COMPATIBILITY
+import vtk
+import sys
+
+from vmtk import vtkvmtk
+from vmtk import vmtkrenderer
+from vmtk import pypes
+
+try:
+    import numpy as np
+except ImportError:
+    raise ImportError('Unable to Import vmtkimagetonumpy module, numpy is not installed')
+
+
+# convenience class for nested disctionaries
+class vividict(dict):
+    def __missing__(self, key):
+        value = self[key] = type(self)()
+        return value
+
+
+class vmtkNumpyToImage(pypes.pypeScript):
+    def __init__(self):
+
+        pypes.pypeScript.__init__(self)
+
+        self.ArrayDict = None
+
+        self.SetScriptName('vmtkNumpyToImage')
+        self.SetScriptDoc('Takes a dictionary of numpy arrays and returns a vtkImageData object.')
+        self.SetInputMembers([
+            ['ArrayDict', 'o', 'dict', 1]])
+        self.SetOutputMembers([
+            ['Image', 'i', 'vtkImageData', 1, '', 'the input image', 'vmtkimagereader']])
+
+    def Execute(self):
+
+        self.Image = vtk.vtkImageData()
+        self.Image.SetDimensions(self.ArrayDict['Dimensions'])
+        self.Image.SetOrigin(self.ArrayDict['Origin'])
+        self.Image.SetSpacing(self.ArrayDict['Spacing'])
+        self.Image.SetExtent((0, self.ArrayDict['Dimensions'][0] - 1,
+                                0, self.ArrayDict['Dimensions'][1] - 1,
+                                0, self.ArrayDict['Dimensions'][2] - 1,))
+
+        self.PrintLog('Converting Numpy Array to vtkImageData')
+        pointDataKeys = self.ArrayDict['PointData'].keys()
+        for key in pointDataKeys:
+            pointDataDType = str(self.ArrayDict['PointData'][key].dtype)
+            if pointDataDType.find('float') != -1:
+                pointDataArray = vtk.vtkFloatArray()
+            if pointDataDType.find('int') != -1:
+                pointDataArray = vtk.vtkIntArray()
+
+            pointDataArray.SetNumberOfComponents(1)
+            pointDataArray.SetName(key)
+
+            flatArray = self.ArrayDict['PointData'][key].ravel(order='F')
+            for pointData in flatArray:
+                pointDataArray.InsertNextValue(pointData)
```

Comment:
If you know how many elements you'll add to `pointDataArray`, it's faster to pre-allocate memory with `pointDataArray.SetNumberOfValues(...)` and the add the values with `pointDataArray.SetValue(i,pointData)`.
`InsertNextValue` is slower because it will likely lead to re-allocations and memcpy's as it grows.